backend/src/datasources/userDatasource.js

```diff
@@ -56,6 +74,8 @@ export class UserDataSource extends DataSource {
 
   async signup (args) {
     const saltRounds = 10
+    await this.getAll()
+
     if (this.users.some((user) => user.email === args.email)) {
       throw new UserInputError('This email already exists')
     }
```

Comment:
👎 E-Mails should at least be checked for an existing '@' symbol inside the string. Also empty strings are still valid.